pkg/healthchecker/health_checker_windows.go

```diff
@@ -37,8 +37,11 @@ func getUptimeFunc(service string) func() (time.Duration, error) {
 		// The powershell command formats the TimeCreated of the event log in RFC1123Pattern.
 		// However, because the time library parser does not recognize the ',' in this RFC1123Pattern format,
 		// it is manually removed before parsing it using the UptimeTimeLayout.
-		getTimeCreatedCmd := "(Get-WinEvent -Logname System | Where-Object {$_.Message -Match '.*(" + service +
-			").*(running).*'} | Select-Object -Property TimeCreated -First 1 | foreach {$_.TimeCreated.ToString('R')} | Out-String).Trim()"
+		getTimeCreatedCmd := `$ProcessId = (Get-WMIObject -Class Win32_Service -Filter "Name='` + service + `'" | Select-Object -ExpandProperty ProcessId);` +
+			`if ([string]::IsNullOrEmpty($ProcessId) -or $ProcessId -eq 0) { (Get-WinEvent -FilterHashtable @{logname='system';id=7036} ` +
```

Comment:
What is id=7036? (in other words, why did this change from the former command, `Get-WinEvent -Logname System | Where-Object ...`?)

Consider explaining this in the comment.

Reply:
I was looking at the events that I wanted and noticed the common id that we can add a filter on it, to unnecessary searches when we do the `Where-Object` command. I will update the comments

Reply:
Thanks. Please put the comment here in the code, rather than (or in addition to) in the PR description, so that readers of the code will understand it easily.

Reply:
Thanks. Please put the comment here in the code, rather than (or in addition to) in the PR description, so that readers of the code will understand it easily.

Reply:
Thanks. Please put the comment here in the code, rather than (or in addition to) in the PR description, so that readers of the code will understand it easily.